internal/services/springcloud/spring_cloud_gateway_sso_resource.go

```diff
@@ -0,0 +1,242 @@
+		},
+
+		"issuer_uri": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"scope": {
+			Type:     pluginsdk.TypeSet,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+	}
+}
+
+func (s SpringCloudGatewaySsoResource) Attributes() map[string]*schema.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (s SpringCloudGatewaySsoResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model SpringCloudGatewaySsoModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.AppPlatform.GatewayClient
+			id, err := parse.SpringCloudGatewayID(model.SpringCloudGatewayId)
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, id.ResourceGroup, id.SpringName, id.GatewayName)
+			if err != nil {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+			if existing.Properties != nil && existing.Properties.SsoProperties != nil {
+				return metadata.ResourceRequiresImport(s.ResourceType(), id)
+			}
+
+			existing.Properties.SsoProperties = &appplatform.SsoProperties{
+				Scope:        utils.ToPtr(model.Scope),
+				ClientID:     utils.String(model.ClientID),
+				ClientSecret: utils.String(model.ClientSecret),
+				IssuerURI:    utils.String(model.IssuerUri),
+			}
+
+			future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.SpringName, id.GatewayName, existing)
+			if err != nil {
+				return fmt.Errorf("creating/updating %s: %+v", id, err)
+			}
+
+			if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+				return fmt.Errorf("waiting for creation/update of %s: %+v", id, err)
+			}
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (s SpringCloudGatewaySsoResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model SpringCloudGatewaySsoModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.AppPlatform.GatewayClient
+			id, err := parse.SpringCloudGatewayID(model.SpringCloudGatewayId)
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, id.ResourceGroup, id.SpringName, id.GatewayName)
+			if err != nil {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			existing.Properties.SsoProperties = &appplatform.SsoProperties{
+				Scope:        utils.ToPtr(model.Scope),
+				ClientID:     utils.String(model.ClientID),
+				ClientSecret: utils.String(model.ClientSecret),
+				IssuerURI:    utils.String(model.IssuerUri),
+			}
```

Comment:
we should be using `d.HasChanges` around these to conditionally patch them only if there's changes